### PR TITLE
move .analyze-cache/ to $HOME/.chef-workstation

### PR DESCRIPTION
**Pull request by afiune**
## Description
Closes https://github.com/chef/chef-analyze/issues/101

Signed-off-by: Salim Afiune <afiune@chef.io>

## Related Issue

https://github.com/chef/chef-analyze/issues/101
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Chore (non-breaking change that does not add functionality or fix an issue)

## Checklist:

- [x] I have read the **CONTRIBUTING** document.
- [ ] I have run the pre-merge tests locally and they pass.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.
- [x] All commits have been signed-off for [the Developer Certificate of Origin](https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco).

### CODE REVIEW

File: go.sum [hunk 1]
```diff
@@ -15,6 +15,8 @@ github.com/chef/go-chef v0.0.0-20191115205148-7ac24fba7a27 h1:XGQlXv6Gwd2+tBBXrp
 github.com/chef/go-chef v0.0.0-20191115205148-7ac24fba7a27/go.mod h1:ft8oU7s0ktse9xzZO4n8wxTz6EKx+V+fuhYxktqXvG8=
 github.com/chef/go-libs v0.0.0-20191113223254-d2a05fa5636d h1:0w673WjR7dQUn1wcNKhMKfQ06GQQvik6uYdcwaWINsg=
 github.com/chef/go-libs v0.0.0-20191113223254-d2a05fa5636d/go.mod h1:aGHwP5PXYdD0hVq+jodILnSqB/b04EhS/XCoqhE2PrA=
+github.com/chef/go-libs v0.0.0-20191216182352-aa2fc405b05f h1:kBujwkOBM7smuh1xCKHrUTdkZbNC0IH30zzDmTNYp8U=
```
Comment (tyler-ball) [inline go.sum]: I hadn't noticed this before - does it keep all known versions of the `go-libs` library?

File: pkg/reporting/cookbooks.go [hunk 2]
```diff
@@ -96,7 +101,18 @@ func (r *CookbookRecord) NumCorrectable() int {
...
+	cbi CookbookInterface, searcher SearchInterface,
+	runCookstyle, onlyUnused bool, workers int,
+) (*CookbooksStatus, error) {
+	wsDir, err := config.ChefWorkstationDir()
+	if err != nil {
+		return nil, err
+	}
+
+	cookbooksDir := filepath.Join(wsDir, analyzeCacheDir, analyzeCookbooksDir)
+	//debug("%s: using cookbooks directory %s", time.Now(), cookbooksDir)
```
Comment (tyler-ball) [inline pkg/reporting/cookbooks.go]: Commented out debug statement - I'm fine with leaving it if you did this intentionally

Comment (afiune) [reply pkg/reporting/cookbooks.go]: I did, I have a bunch that I wanna tackle sooner rather than later. 🤞

File: go.sum (hunk 1 above)
Comment (afiune) [reply go.sum]: yeah, It is kinda odd and confusing, we can clean it up by running `go mod tidy`, I will try to do that.

File: pkg/reporting/cookbooks_test.go [hunk 3]
```diff
@@ -234,7 +234,7 @@ func TestCookbooks(t *testing.T) {
 	c, err := subject.NewCookbooks(
 		newMockCookbook(cookbookList, nil, nil),
 		makeMockSearch(mockedNodesSearchRows(), nil),
-		false,
+		true,
```
Comment (marcparadise) [inline pkg/reporting/cookbooks_test.go]: This is because we weren't testing the extended data?

File: pkg/reporting/cookbooks_test.go [hunk 4]
```diff
@@ -257,6 +257,9 @@ func TestCookbooks(t *testing.T) {
 				default:
 					t.Fatal("unexpected cookbook name")
 				}
+				// TODO @afiune uncomment when CookbookMock DownloadTo() creates a local dir
```
Comment (marcparadise) [inline pkg/reporting/cookbooks_test.go]: Is there a way to test this that could avoid having the the mock touch the local filesystem?

File: pkg/reporting/cookbooks_test.go (hunk 3 above)
Comment (afiune) [reply pkg/reporting/cookbooks_test.go]: Yes.

File: pkg/reporting/cookbooks_test.go (hunk 4 above)
Comment (afiune) [reply pkg/reporting/cookbooks_test.go]: I think that it is safe to create the directory once all our tests are running on temp directories, we should fix that with https://github.com/chef/chef-analyze/issues/57

**Pull request merged**